lib/std/array_list.zig

```diff
@@ -87,6 +87,17 @@ pub fn ArrayListAligned(comptime T: type, comptime alignment: ?u29) type {
             };
         }
 
+        /// ArrayList takes ownership of the passed in slice. The slice must have been
+        /// allocated with `allocator`.
+        /// Deinitialize with `deinit` or use `toOwnedSlice`.
+        pub fn fromOwnedSliceSentinel(allocator: Allocator, comptime sentinel: T, slice: [:sentinel]T) Self {
+            return Self{
+                .items = @ptrCast([*]T, slice.ptr)[0..slice.len],
```

Comment:
What is this line trying to do? The `ptrCast` is a no-op (it's already `[*]T`) and the slice doesn't do anything. Did you mean to do `slice.ptr[0..slice.len + 1]`, which would extend the length to include the sentinel?

Reply:
Mmm that's a good callout, the ptrCast is some sloppiness.

The intended outcome here is to have the items.len be the length of the slice _excluding_ the sentinel, but including the sentinel in the capacity.

So the .items initializer can just be:

```zig
                .items = slice[0..slice.len],
```

Reply:
Oh, you don't need to do anything fancy then, just use `.items = slice`

The length does not include the sentinel

Reply:
🤦‍♂️ D'oh right, I should stop replying to PR reviews right before bed. Thank you 🙂